ci_framework/plugins/module_utils/net_map/ip_pools.py

```diff
@@ -1,15 +1,18 @@
 import ipaddress
 import typing
 
-from ansible_collections.cifmw.general.plugins.module_utils.networking_mapping import (
-    exceptions,
-    networking_definition,
-)
+try:
```

Comment:
This import hack here is my only concern. I'm ok with the entire PR, but for me, touching the "product" code just because we need it to build docs is an smell like when people touch the code to make unit tests. Something is not properly done. I know that in this case the limitation is in the way ansible handles imports, that need to be done using that ansible_collections.<namespace>.<collection> pattern, but, why can't us play with the PYTHONPATH to make it work without touching the code itself?

Reply:
seems really related to the PYTHONPATH in fact: unless we nudge the content to something matching the `ansible_collections.foo.bar.super.long.name` while building the doc, we're a bit stuck with that.

I can add a comment explaining why this is done.
We'd need to mock the actual module in sphinxdoc - or "proxy" it... AFAIK we can do it with python, so maybe we can do it in the conf.py?

Reply:
Can be done there, but I was thinking on a simpler way, create an directory structure that matches the `ansible_collections/cifmw/general` pattern, symlink that general dir to the `ci_framework` directory and add the parent dir of the created `ansible_collections` to the PYTHONPATH. It should do the job.